sunpy/coordinates/transformations.py

```diff
@@ -127,6 +127,14 @@ def _observers_are_equal(obs_1, obs_2, string_ok=False):
             obs_1.obstime == obs_2.obstime)
 
 
+def _check_observer_defined(frame):
+    if frame.observer is None:
+        raise ConvertError("This transformation cannot be performed with observer=None.")
```

Comment:
Given you call this function on input and output frames in the transforms would it be wise to specify which frame is missing the observer here?

Reply:
Done